### PR TITLE
chore: add explicit return types to tradeQuoteSlice selectors

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

Safety first. Add explicit return types to tradeQuoteSlice selectors.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

Minimal, no un-feature flagged changes.

## Testing

App should show no regressions, though nothing specific to test.

### Engineering

☝️

### Operations

☝️

## Screenshots (if applicable)

N/A

### CODE REVIEW

**Comment by 0xApotheosis** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4797** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4797" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4797?utm_source=stack-comment).

File: src/state/slices/tradeQuoteSlice/selectors.ts [hunk 1]
```diff
@@ -17,25 +21,20 @@ import {
 const selectTradeQuoteSlice = (state: ReduxState) => state.tradeQuoteSlice
 
 // TODO(apotheosis): Cache based on quote ID
-export const selectSelectedQuote = createDeepEqualOutputSelector(
-  selectTradeQuoteSlice,
-  swappers => swappers.quote,
-)
+export const selectSelectedQuote: Selector<ReduxState, TradeQuote | undefined> =
```
Comment (gomesalexandre) [inline src/state/slices/tradeQuoteSlice/selectors.ts:24]: Interesting type assertion, do we need this if `selectTradeQuoteSlice` is typed? And if we want to strongly type the return type, can't we type it on the line below instead i.e in the callback?

**Review by gomesalexandre — APPROVED** [review]
Conceptually LGTM, one q. regarding the use of `Selector<>` type assertion instead of typing the callback, but conceptually happy with it

File: src/state/slices/tradeQuoteSlice/selectors.ts (hunk 1 above)
Comment (0xApotheosis) [reply src/state/slices/tradeQuoteSlice/selectors.ts:24]: I'm not sure if this question is directed at this particular usage, or this PR as a whole, but the key intent of the PR is that these are _not_ type assertions (weakening type checking), they are type _annotations_ to ensure that what we are expecting to be returned by each selector is enforced at compile time.

**Pull request merged**